### PR TITLE
SCP-872: Automatically process contract requests

### PR DESCRIPTION
**Pull request by j-mueller**
* Starting the PAB with `all-servers` also starts a thread that processes the contract outboxes in a configurable interval. 
* `processContractOutboxes` handles more than one request per iteration

### CODE REVIEW

File: plutus-scb/plutus-scb.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ dbConfig:
 
 scbWebserverConfig:
   baseUrl: http://localhost:8080
+  staticDir: plutus-scb-client/dist
```
Comment (michaelpj) [inline plutus-scb/plutus-scb.yaml:7]: Should this have changed?

File: plutus-scb/src/Plutus/SCB/Core/ContractInstance.hs [hunk 2]
```diff
@@ -298,12 +303,20 @@ respondtoRequests ::
...
-    => RequestHandler effs ContractSCBRequest ContractResponse
+    => MaxIterations
+    -> RequestHandler effs ContractSCBRequest ContractResponse
     -> Eff effs ()
-respondtoRequests handler = do
+respondtoRequests (MaxIterations mi) handler = do
     contractStates <- runGlobalQuery (Query.contractState @t)
     let state = fmap (hooks . csCurrentState) contractStates
-    itraverse_ (runRequestHandler @t handler) state
+    flip itraverse_ state $ \instanceId requests ->
```
Comment (michaelpj) [inline plutus-scb/src/Plutus/SCB/Core/ContractInstance.hs:312]: `ifor_`

File: plutus-scb/app/Main.hs [hunk 3]
```diff
@@ -370,9 +377,12 @@ runCliCommand _ _ _ (ReportContractHistory uuid) = do
...
     Core.processContractInbox @ContractExe (ContractInstanceId uuid)
-runCliCommand _ _ _ ProcessAllContractOutboxes = do
-    logInfo ProcessAllOutboxesMsg
-    Core.processAllContractOutboxes @ContractExe
+runCliCommand _ Config{requestProcessingConfig} _ ProcessAllContractOutboxes = do
+    let RequestProcessingConfig{requestProcessingInterval} = requestProcessingConfig
+    logInfo $ ProcessAllOutboxesMsg requestProcessingInterval
+    forever $ do
+        _ <- liftIO . threadDelay . fromIntegral $ toMicroseconds requestProcessingInterval
+        handleError @SCBError (Core.processAllContractOutboxes @ContractExe Trace.defaultMaxIterations) (logError . ContractExeSCBError)
```
Comment (michaelpj) [inline plutus-scb/app/Main.hs:385]: It seems wrong that this depends on the trace module, perhaps the default should move somewhere else?

File: plutus-scb/plutus-scb.yaml (hunk 1 above)
Comment (j-mueller) [reply plutus-scb/plutus-scb.yaml:7]: This is present in `plutus-scb/plutus-scb.yaml.sample` but not here. The real solution is probably to delete `plutus-scb.yaml` and add it to gitignore...

File: plutus-scb/app/Main.hs (hunk 3 above)
Comment (j-mueller) [reply plutus-scb/app/Main.hs]: I moved it to `Language.Plutus.Contract.Trace.RequestHandler`

File: plutus-scb/plutus-scb.yaml (hunk 1 above)
Comment (j-mueller) [reply plutus-scb/plutus-scb.yaml]: .. which I just did :-)

**Pull request merged**